lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,368 @@
+defmodule Gringotts.Gateways.Mercadopago do
+  @moduledoc """
+  [mercadopago][home] gateway implementation.
+
+  For reference see [mercadopago documentation][docs].
+
+  The following features of mercadopago are implemented:
+
+  | Action                       | Method        |
+  | ------                       | ------        |
+  | Pre-authorize                | `authorize/3` |
+  | Capture                      | `capture/3`   |
+  | Purchase                     | `purchase/3`  |
+  | Reversal                     | `void/2`      |
+  | Refund                       | `refund/3`    |
+
+  [home]: https://www.mercadopago.com/
+  [docs]: https://www.mercadopago.com.ar/developers/en/api-docs/
+
+  ## The `opts` argument
+
+  Most `Gringotts` API calls accept an optional `keyword` list `opts` to supply
+  optional arguments for transactions with the mercadopag. The following keys are supported:
+
+  | Key                      | Remark                                                                                  |
+  | ----                     | ---                                                                                     |
+  | `email`                  | Email of the customer. Type - string                                                    |
+  | `order_id`               | Order id issued by the merchant. Type- integer                                          |
+  | `customer_id`            | Unique customer id issued by the gateway. For new customer it must skipped. Type- string|
+  | `order_type`             | `"mercadopago"` or `"mercadolibre"` as per the order. Type- string                      |
+  | `installments`           | No of installments for payment. Type- integer                                           |
+
```

Comment:
@mandarvaze  Add PCI and Non PCI section and implementation with info.

```

### I'm not PCI-DSS compliant
  No worries, both `authorize/3` and `purchase/3` accept a
  "payment-source-identifier" (a `string`) instead of a `CreditCard.t`
  struct. You'll have to generate this identifier using [Stripe.js and
  Elements][stripe-js] client-side.
  ### I'm PCI-DSS compliant
  In that case, you need not use [Stripe.js or Elements][stripe-js] and can
  directly accept the client's card info and pass the `CreditCard.t` struct to
  this module's functions.
  [stripe-js]: https://stripe.com/docs/sources/cards
```